Add packages brasero and soundjuicer

bdimcheff opened the pull request:
Packaged up brasero and sound-juicer, a gnome cd burner and ripper, respectively.

bdimcheff committed:
add brasero, a gnome cd/dvd burner

bdimcheff committed:
add sound juicer, a gnome cd ripper

bdimcheff commented on pkgs/applications/audio/sound-juicer/default.nix:
```diff
@@ -0,0 +1,47 @@
+  preFixup = ''
+    for f in $out/bin/* $out/libexec/*; do
+      wrapProgram "$f" \
+        --prefix XDG_DATA_DIRS : "${gnome3.gnome_themes_standard}/share:$XDG_ICON_DIRS:$GSETTINGS_SCHEMAS_PATH" \
```
is there more stuff that needs to be wrapped for gnome apps?  I copied this from some other gnome apps, but I'm not sure if other envs are necessary.

offlinehacker replied:
Have you tested, and does it work?

bdimcheff replied:
yeah I was able to successfully rip and burn cds with these two apps, but I dunno if there's some gnome functionality that is missing due to lack of proper env vars or something... I don't actually use gnome itself, so I'm not 100% sure what to look for.

bdimcheff commented on pkgs/tools/cd-dvd/brasero/remove-symlink-check.patch:
```diff
@@ -0,0 +1,29 @@
+-	 * the symlink with GIO. The problem is, when the symlink pointed to
+-	 * another symlink, then GIO didn't follow that other symlink. And in
+-	 * the end it didn't work. So forbid all symlink. */
+-	if (g_file_test (prog_path, G_FILE_TEST_IS_SYMLINK)) {
```
This was checking to make sure apps in the path are not symlinks.  That doesn't work so well here... ;)

offlinehacker replied:
Makes sense

bdimcheff commented:
hmm `error: attribute ‘gcc’ missing, at "/home/travis/.nox/nixpkgs/pkgs/build-support/cc-wrapper/default.nix":174:38` doesn't seem like something I did?

offlinehacker commented on pkgs/applications/audio/sound-juicer/default.nix:
```diff
@@ -0,0 +1,47 @@
+{ stdenv, fetchurl, pkgconfig, gtk3, intltool, itstool, libxml2, brasero, libcanberra_gtk3, gnome3, gst_all_1, libmusicbrainz5, libdiscid, isocodes, makeWrapper }:
```
This line could be split, but it's not important, just to let you know when/if you will be updating these packages.

bdimcheff replied:
sure, I'll wrap lines

bdimcheff committed:
wrap long lines in sound-juicer

bdimcheff commented:
@offlinehacker I think this is all set now

bdimcheff commented:
hmm actually I think I fixed the gsettings errors I saw, one sec

bdimcheff committed:
add dconf dependencies to sound-juicer and brasero

If you have services.gnome3.enabled = true, you shouldn't need this, but
if you don't, you'll need this dconf env var to be able to save settings
in these apps.

bdimcheff commented:
OK so 93228a0 should make this work with gsettings/dconf, even outside of a gnome session.

lethalman commented:
Built locally, thanks.
